CMake fails to detect ABI info with Zig cc/c++.

### Zig Version

0.14.0-dev.144+a31fe8aa3

### Steps to Reproduce and Observed Behavior

CMake passes `-v` linker flag during ABI info check causing Zig to fail with an error:

```
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_b0cec.dir/link.txt --verbose=1
/home/user/zig/master/zig c++   -v -Wl,-v CMakeFiles/cmTC_b0cec.dir/CMakeCXXCompilerABI.cpp.o -o cmTC_b0cec
error: unsupported linker arg: -v
make[1]: *** [CMakeFiles/cmTC_b0cec.dir/build.make:100: cmTC_b0cec] Error 1
make[1]: Leaving directory '/home/user/Projects/zig-test/build/CMakeFiles/CMakeScratch/TryCompile-rryASJ'
make: *** [Makefile:127: cmTC_b0cec/fast] Error 2
```

While clang has no problem with this:

```
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_505f3.dir/link.txt --verbose=1
/usr/bin/clang  -v -Wl,-v CMakeFiles/cmTC_505f3.dir/CMakeCXXCompilerABI.cpp.o -o cmTC_505f3
clang version 17.0.6
Target: x86_64-pc-linux-gnu
...
```

Steps to reproduce:

Try to configure this CMake project:

```
cmake_minimum_required(VERSION 3.19)

project(zig_test LANGUAGES CXX C)
```

With zig cc/c++: `cmake . -DCMAKE_CXX_COMPILER=zig -DCMAKE_CXX_FLAGS=c++ -DCMAKE_C_COMPILER=zig -DCMAKE_C_FLAGS=cc`

You will observe these status messages: `Detecting CXX compiler ABI info - failed` and `Detecting C compiler ABI info - failed`
While it's not an error, it's easy to ignore. But without ABI info, CMake loses a lot of information that projects can depend on (a notable example is GNUInstallDirs, which depends on generated info to detect the LIBDIR), such as `CMAKE_SIZEOF_VOID_P`, `CMAKE_C_COMPILER_ABI`, `CMAKE_C_BYTE_ORDER` and many others.

Interestingly, the plain `-v` works as intended, but linker one fails. 


### Expected Behavior

`zig c++ -Wl,-v` works.

--- a/src/main.zig
+++ b/src/main.zig
@@ -2475,6 +2475,8 @@ fn buildOutputType(
                         fatal("unable to parse /version '{s}': {s}", .{ arg, @errorName(err) });
                     };
                     have_version = true;
+                } else if (mem.eql(u8, arg, "-v")) {
+                    try std.io.getStdOut().writeAll("zig ld " ++ build_options.version ++ "\n");
                 } else if (mem.eql(u8, arg, "--version")) {
                     try std.io.getStdOut().writeAll("zig ld " ++ build_options.version ++ "\n");
                     process.exit(0);

--- a/src/main.zig
+++ b/src/main.zig
@@ -2475,6 +2475,8 @@ fn buildOutputType(
                         fatal("unable to parse /version '{s}': {s}", .{ arg, @errorName(err) });
                     };
                     have_version = true;
+                } else if (mem.eql(u8, arg, "-V")) {
+                    warn("ignoring request for supported emulations: unimplemented", .{});
                 } else if (mem.eql(u8, arg, "-v")) {
                     try std.io.getStdOut().writeAll("zig ld " ++ build_options.version ++ "\n");
                 } else if (mem.eql(u8, arg, "--version")) {

